### ISSUE TITLE
Setting up in existing project

### ISSUE THREAD

**Issue by gokul1099** [mrzachnugent/react-native-reusables#131]
I have existing react native project, How can i use react-native-reuasble? How can i do the initial configuration to use the cli later to add needed components. I have gone through the docs, looks like some sub sections in installation sections are missing.

**Comment by mrzachnugent** [comment]
Hi @gokul1099, 

On the [Initial Setup](https://rnr-docs.vercel.app/getting-started/initial-setup/) page, you can select the 'Manual Installation" tab.

Then, you can use the CLI to add components to your project. You can use `npx @react-native-reusables/cli@latest add`, then it will prompt you with the type of project you are working with (universal or native-only), with the location of your `/components` and `/lib` folders, then you can select the components you want to add to your project. It will, then, add the files and dependencies (if any).

Is is possible that there are sections that are missing information. Could you please provide the exact cases please? That way I could fix it

**Comment by itsyashsahu** [comment]
Hi @mrzachnugent 
I am trying to use the official repo starter base to make the universal react app. 
I am following the steps mentioned in the docs but still the metro bulid gets stuck without throwing an error. 

Can you please help me with this ?

**Comment by rickychan0611** [comment]
> Hi @mrzachnugent I am trying to use the official repo starter base to make the universal react app. I am following the steps mentioned in the docs but still the metro bulid gets stuck without throwing an error.
> 
> Can you please help me with this ?

I have the same problem.

**Comment by mrzachnugent** [comment]
Hey @itsyashsahu @rickychan0611,

I am going to try it on my side and document the steps I am taking here.

1. `git clone https://github.com/mrzachnugent/react-native-reusables.git`
2. `cp -R react-native-reusables/apps/starter-base/ ./my-project`
3. `cd my-project && npm install`
4. `npm run ios`
5. I ran into this error which seems to be an npm issue: "Error: EMFILE: too many open files, watch" so I deleted the `node_modules` and ran `npm install` once again
6. Then it worked.

Let me know if you this fixes your issue.

<img width="347" alt="Screenshot 2024-04-19 at 4 46 04 PM" src="https://github.com/mrzachnugent/react-native-reusables/assets/63797719/9f66dce1-2632-4f69-ac2f-91c9170b4c9a">

**Comment by itsyashsahu** [comment]
@mrzachnugent I am doing the same steps 

1. git clone https://github.com/mrzachnugent/react-native-reusables.git
2. cp -R react-native-reusables/apps/starter-base/ ./my-project
3. cd my-project && npm install
4. npm run web 

then it gets stucked at a point. I tried it multiple times

**Comment by Alejandro-M-Cruz** [comment]
Hi, I'm having the same issue. It works fine on Android, but when running on web it gets stuck while bundling. No error message or anything, the terminal just freezes. I have tried with both `npm` and `yarn`. At least in my case, the issue is only on Windows 10. I have tried on macOS and everything works fine. 

System details:
- Windows 10
- NodeJS v20.10.0
- npm 10.5.0
- yarn 1.22.22
- expo 50.0.17

**Comment by rickychan0611** [comment]
It is indeed a problem within nativewind v4. I discovered a pull request in Nativewind's repository and tested it out. I can confirm that it no longer gets stuck on Windows with the correction. However, the pull request has not been merged yet.
https://github.com/marklawlor/nativewind/pull/860/commits/59847eb4358efde93cf93a55b5a60812f7661917

**Comment by a0m0rajab** [comment]
Related PR link: https://github.com/marklawlor/nativewind/pull/860

**Comment by ndarcode** [comment]
any updates?

**Comment by mrzachnugent** [comment]
> any updates?

For updates, you can check the pull request on NativeWind's Github repo: https://github.com/marklawlor/nativewind/pull/860

When it is solved, I will close this issue.

**Comment by SHIXUNXUN** [comment]
same bug

**Comment by baptlma** [comment]
same issue

**Comment by mrzachnugent** [comment]
> is going nativewind a good route at this time?

Yes, I am betting on it. The sole maintainer had taken a break but he now has a team behind him and they are working hard. Here's the latest update https://x.com/danstepanov/status/1806107770799968673?s=46&t=tpxUSfNj59WG9Y3VDKw5sQ

**Comment by a0m0rajab** [comment]
Thanks for the update Zach, they had a simple announcement on the v4 issue just for future reference: https://github.com/nativewind/nativewind/issues/600

**Comment by guibzo** [comment]
I uploaded a temporary solution to this [here](https://github.com/nativewind/nativewind/pull/860#issuecomment-2243046234)

**Comment by tripolskypetr** [comment]
@guibzo Work like charm, just used It

**Comment by tripolskypetr** [comment]
I also making a research about running the best `react-native` ui kit starter app. Found a little mistake in `starter-base`. After the nativewind fix, the build will continue failing

See https://github.com/mrzachnugent/react-native-reusables/pull/212

@mrzachnugent

**Comment by manssorr** [comment]
Check this patch: https://github.com/nativewind/nativewind/pull/860#issuecomment-2371839082
